src/Content/Version.php

```diff
@@ -146,23 +166,105 @@ public function move(
+			fromLanguage: Language::ensure($fromLanguage),
 			toVersionId: $toVersionId,
-			toLanguage: $this->language($toLanguage)
+			toLanguage: Language::ensure($toLanguage)
 		);
 	}
 
+	/**
+	 * Prepare fields to be written by removing unwanted fields
+	 * depending on the language or model and by cleaning the field names
+	 */
+	protected function prepareFieldsBeforeWrite(array $fields, Language $language): array
+	{
+		// convert all field names to lower case
+		$fields = $this->convertFieldNamesToLowerCase($fields);
+
+		// make sure to store the right fields for the model
+		$fields = $this->model->contentFileData($fields, $language);
+
+		// the default language stores all fields
+		if ($language->isDefault() === true) {
+			return $fields;
+		}
+
+		// remove all untranslatable fields
+		foreach ($this->model->blueprint()->fields() as $field) {
+			if (($field['translate'] ?? true) === false) {
+				unset($fields[strtolower($field['name'])]);
+			}
+		}
+
+		// remove UUID for non-default languages
+		unset($fields['uuid']);
+
+		return $fields;
+	}
+
+	/**
+	 * Make sure that reading from storage will always
+	 * return a usable set of fields with clean field names
+	 */
+	protected function prepareFieldsAfterRead(array $fields, Language $language): array
+	{
+		return $this->convertFieldNamesToLowerCase($fields);
+	}
+
 	/**
 	 * Returns the stored content fields
 	 *
-	 * @return array<string, string>
+	 * @return array<string, string>|null
+	 */
+	public function read(Language|string $language = 'default'): array|null
+	{
+		$language = Language::ensure($language);
+
+		try {
+			$fields = $this->model->storage()->read($this->id, $language);
+			$fields = $this->prepareFieldsAfterRead($fields, $language);
+			return $fields;
+		} catch (NotFoundException) {
+			return null;
+		}
+	}
+
+	/**
+	 * Replaces the content of the current version with the given fields
+	 *
+	 * @param array<string, string> $fields Content fields
 	 *
 	 * @throws \Kirby\Exception\NotFoundException If the version does not exist
 	 */
-	public function read(Language|string $language = 'default'): array
+	public function replace(array $fields, Language|string $language = 'default'): void
 	{
 		$this->ensure($language);
-		return $this->model->storage()->read($this->id, $this->language($language));
+
+		$language = Language::ensure($language);
+
+		$this->model->storage()->update(
+			versionId: $this->id,
+			language: $language,
+			fields: $this->prepareFieldsBeforeWrite($fields, $language)
+		);
+	}
+
+	/**
+	 * Convenience wrapper around ::create, ::replace and ::update.
+	 */
+	public function save(
+		array $fields,
+		Language|string $language = 'default',
+		bool $overwrite = false
+	): void {
+		match (true) {
+			$this->exists($language) === false
+				=> $this->create($fields, $language),
+			$overwrite === true
+				=> $this->replace($fields, $language),
+			default
+			=> $this->update($fields, $language)
+		};
```

Comment:
`Suggestion` to increase readability:

```suggestion
		if ($this->exists($language) === false) {
			$this->create($fields, $language);
			return;
		}

		if ($overwrite === true) {
			$this->replace($fields, $language);
			return;
		}

		$this->update($fields, $language);
```